### PR TITLE
[tests-only][full-ci] Improve Then step in apiShareReshareToShares1 suite

### PR DESCRIPTION
**Pull request by amrita-shrestha**
## Description
This PR adds `Then` steps to existing API test scenarios to better check the results of the `When` steps.

**Suites covered:**
- `apiShareReshareToShares1`

## Related Issue

- Part of https://github.com/owncloud/core/issues/39512

## How Has This Been Tested?

- Locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/apiShareReshareToShares1/reShare.feature [hunk 1]
```diff
@@ -50,11 +50,9 @@ Feature: sharing
     And user "Alice" has shared file "/textfile0.txt" with user "Brian" with permissions "share,read"
     And user "Brian" has accepted share "/textfile0.txt" offered by user "Alice"
     When user "Brian" shares file "/Shares/textfile0.txt" with user "Carol" with permissions "share,read" using the sharing API
-    Then the OCS status code should be "<ocs_status_code>"
-    And the HTTP status code should be "200"
-    When user "Carol" accepts share "/textfile0.txt" offered by user "Brian" using the sharing API
-    Then the OCS status code should be "<ocs_status_code>"
-    And the HTTP status code should be "200"
+    And user "Carol" accepts share "/textfile0.txt" offered by user "Brian" using the sharing API
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiShareReshareToShares1/reShare.feature:53]:
```suggestion
    And user "Carol" should be able to accept pending share "/textfile0.txt" offered by user "Brian"
```

Comment (amrita-shrestha) [reply tests/acceptance/features/apiShareReshareToShares1/reShare.feature:53]: @kiranparajuli589 Do you mean adding new step ` Then user "Carol" should be able to accept pending share "/textfile0.txt" offered by user "Brian"`

**Comment by SwikritiT** [comment]
@amrita-shrestha if this is ready for review maybe you can open it up from draft

File: tests/acceptance/features/apiShareReshareToShares1/reShare.feature (hunk 1 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/apiShareReshareToShares1/reShare.feature:53]: yes exactly

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1:

**Review by SwikritiT — COMMENTED** [review]
LGTM

File: tests/acceptance/features/apiShareReshareToShares1/reShare.feature (hunk 1 above)
Comment (individual-it) [reply tests/acceptance/features/apiShareReshareToShares1/reShare.feature]: The question is accepting the share part of the sharing-action or part of the expected result. If its part of the action it should be a When step

Comment (kiranparajuli589) [reply tests/acceptance/features/apiShareReshareToShares1/reShare.feature]: here, in most of the scenarios, we're checking whether the user is able to reshare. so `accept-share` part from the receiver side is not the main action imo.

**Review by individual-it — APPROVED** [review]

**Pull request merged**